### PR TITLE
Test Cases for ApiDAOImpl and PolicyDAOimpl

### PR DESCRIPTION
**Pull request by dinusha92**
### Proposed changes in this pull request

test cases for ApiDAOImpl
test cases for PolicyDAOImpl
Analyzer module policy modifed

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImplIT.java [hunk 1]
```diff
@@ -1683,4 +1678,52 @@ public void testGetResourcesOfApi() throws Exception {
...
+        Assert.assertEquals(compositeAPI.getContext(), summaryAPI.getContext());
+
+        //Composite APIs retrieving
+        List<CompositeAPI> compositeAPIS = apiDAO.getCompositeAPIs(new HashSet<>(), compositeAPI.getProvider(), 4, 4);
+        Assert.assertEquals(compositeAPI.getId(), compositeAPIS.get(0).getId());
+
+        //Delete Composite API
+        apiDAO.deleteCompositeApi(compositeAPI.getId());
+        Assert.assertNull(apiDAO.getCompositeAPI(compositeAPI.getId()));
+
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImplIT.java]: Remove the line

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java [hunk 2]
```diff
@@ -83,6 +96,15 @@ public void testFingerprintAfterUpdatingApplicationPolicy() throws Exception {
...
                 + policy.getPolicyName());
+        //Test for exception uuid null
+        updatedPolicy.setUuid(null);
+        try {
+            policyDAO.updateApplicationPolicy(updatedPolicy);
+        } catch (APIMgtDAOException ex) {
+            Assert.assertEquals(ex.getMessage(),
+                    "Policy uuid is not found, unable to update policy: " + updatedPolicy.getPolicyName());
+        }
+
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: Remove unnecessary line

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java [hunk 3]
```diff
@@ -103,6 +125,14 @@ public void testFingerprintAfterUpdatingSubscriptionPolicy() throws Exception {
         Assert.assertNotNull(fingerprintAfterUpdatingPolicy);
         Assert.assertNotEquals(fingerprintBeforeUpdatingPolicy, fingerprintAfterUpdatingPolicy, TestUtil.printDiff
                 (fingerprintBeforeUpdatingPolicy, fingerprintAfterUpdatingPolicy));
+
+        updatedPolicy.setUuid(null);
+        try {
+            policyDAO.updateSubscriptionPolicy(updatedPolicy);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: Should we add Assert.fail() here?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java [hunk 4]
```diff
@@ -83,6 +96,15 @@ public void testFingerprintAfterUpdatingApplicationPolicy() throws Exception {
         Assert.assertNotEquals(fingerprintBeforeUpdatingPolicy, fingerprintAfterUpdatingPolicy, "Policy "
                 + "fingerprint expected to be different before and after updating for policy: "
                 + policy.getPolicyName());
+        //Test for exception uuid null
+        updatedPolicy.setUuid(null);
+        try {
+            policyDAO.updateApplicationPolicy(updatedPolicy);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: Should we add Assert.fail() here?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java [hunk 5]
```diff
@@ -62,6 +67,14 @@ public void testFingerprintAfterUpdatingAPIPolicy() throws Exception {
         Assert.assertNotEquals(fingerprintBeforeUpdatingPolicy, fingerprintAfterUpdatingPolicy, "Policy "
                 + "fingerprint expected to be different before and after updating for policy: "
                 + policy.getPolicyName());
+        updatedAPIPolicy.setUuid(null);
+        //test for exception
+        try {
+            policyDAO.updateApiPolicy(updatedAPIPolicy);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: Should we add Assert.fail() here?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java [hunk 6]
```diff
@@ -293,24 +345,179 @@ public void testAddGetDeleteBlockConditions() throws Exception {
...
+        ApplicationPolicy applicationPolicy = SampleTestObjectCreator.createDefaultApplicationPolicy();
+        SubscriptionPolicy subscriptionPolicy = SampleTestObjectCreator.createDefaultSubscriptionPolicy();
+        policyDAO.addApiPolicy(apiPolicy);
+        policyDAO.addApplicationPolicy(applicationPolicy);
+        policyDAO.addSubscriptionPolicy(subscriptionPolicy);
+        Set<PolicyValidationData> policyValidationData = policyDAO.getAllPolicies();
+        Assert.assertTrue(policyValidationData.size() == size + 3);
+    }
+
+
```
Comment (dakshika) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: Remove unwanted lines

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImplIT.java (hunk 1 above)
Comment (dinusha92) [reply components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImplIT.java]: This has been fixed now.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java (hunk 6 above)
Comment (dinusha92) [reply components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java]: This is fixed with https://github.com/wso2/carbon-apimgt/pull/5147

**Pull request merged**